sssd_test_framework/roles/keycloak.py

```diff
@@ -126,12 +129,66 @@ def add(
 
         return self
 
-    def delete(self) -> None:
+    def delete(
+        self,
+    ) -> SSHProcessResult:
```

Comment:
What is the intention behind this?

Reply:
I think this was left over from an earlier update to return the result to check the error returned.   I have since changed to use this:

```
    with pytest.raises(SSHProcessError, match="User couldn't be deleted"):
        kcuser1.delete()
```

I left this to return the result if needed later but, TBH I can't think of a reason now.   The kcadm delete command typically returns nothing except on failure which I can check like this.

Is checking the raised error like above a good way to accomplish this?   So I can change this back to returning None.  Or would I be better off returning the SSHProcessResult and processing it that way?

Reply:
It really depends on what you want to accomplish this.

Current implementation of `kcadm` raises error on `rc != 0`.

* If you want to assert that a command failed, then you don't need to return the result because kcadm will raise `SSHProccessError` anyway.
* If you want to check something in the result of successful run then you need to return the result.

Another way would be to set `raise_on_error=False` and return the result but then you always have to assert the return code by yourself (`assert result.rc == 0`).

From what you describe, I'd switch back to return `None` and rely on the exception. With that, `pytest.raises` is a good way to assert exceptions.

Reply:
ok, reverting this back to the original returning None.